### DISCUSSION TITLE
Understanding the meshopt_simplify function.

### DISCUSSION THREAD

**Discussion by Zeta48** [zeux/meshoptimizer#510] [Q&A] [answered]
I am looking for a library that can simplify meshes to reduce their memory footprint. The algorithm should preserve texture coordinates.
Meshoptimizer seems to do so. But when I look at the meshopt_simplify function it look like only the index buffer is reduced, while the heavy vertex buffer stays the same. Am i wrong? Is there a way to reduce the vertex buffer as well?

**Comment by zeux** [comment, marked as answer]
The documentation in README says:

> This library provides two simplification algorithms that reduce the number of triangles in the mesh. Given a vertex and an index buffer, they generate a second index buffer that uses existing vertices in the vertex buffer. This index buffer can be used directly for rendering with the original vertex buffer (preferably after vertex cache optimization), or a new compact vertex/index buffer can be generated using meshopt_optimizeVertexFetch that uses the optimal number and order of vertices.

The same note is reflected in the comment above the function definition:

>  * If the original vertex data isn't required, creating a compact vertex buffer using meshopt_optimizeVertexFetch is recommended.

**Reply by Zeta48** [reply]
So first meshopt_simplify followed by meshopt_optimizeVertexFetch.
Thanks for the good work!
